pkg/rewards/2_goldStakerRewardAmounts.go

```diff
@@ -107,26 +108,38 @@ staker_operator_total_tokens AS (
     END as total_staker_operator_payout
   FROM staker_proportion
 ),
--- Calculate the token breakdown for each (staker, operator) pair
+-- Include the operator_avs_split_snapshots table
+operator_avs_splits_cte AS (
+  SELECT
+    operator,
+    avs,
+    snapshot,
+    split
+  FROM operator_avs_split_snapshots
+),
+-- Calculate the token breakdown for each (staker, operator) pair with dynamic split logic
+-- If no split is found, default to 1000 (10%)
 token_breakdowns AS (
-  SELECT *,
+  SELECT sot.*,
     CASE
       WHEN snapshot < @amazonHardforkDate AND reward_submission_date < @amazonHardforkDate THEN
-        cast(total_staker_operator_payout * 0.10 AS DECIMAL(38,0))
+        cast(total_staker_operator_payout * COALESCE(oas.split, 1000) / 10000.0 AS DECIMAL(38,0))
       WHEN snapshot < @nileHardforkDate AND reward_submission_date < @nileHardforkDate THEN
-        (total_staker_operator_payout * 0.10)::text::decimal(38,0)
+        (total_staker_operator_payout * COALESCE(oas.split, 1000) / 10000.0)::text::decimal(38,0)
       ELSE
-        floor(total_staker_operator_payout * 0.10)
+        floor(total_staker_operator_payout * COALESCE(oas.split, 1000) / 10000.0)
     END as operator_tokens,
     CASE
       WHEN snapshot < @amazonHardforkDate AND reward_submission_date < @amazonHardforkDate THEN
-        total_staker_operator_payout - cast(total_staker_operator_payout * 0.10 as DECIMAL(38,0))
+        total_staker_operator_payout - cast(total_staker_operator_payout * COALESCE(oas.split, 1000) / 10000.0 AS DECIMAL(38, 0))
       WHEN snapshot < @nileHardforkDate AND reward_submission_date < @nileHardforkDate THEN
-        total_staker_operator_payout - ((total_staker_operator_payout * 0.10)::text::decimal(38,0))
+        total_staker_operator_payout - ((total_staker_operator_payout * COALESCE(oas.split, 1000) / 10000.0)::text::decimal(38, 0))
       ELSE
-        total_staker_operator_payout - floor(total_staker_operator_payout * 0.10)
+        total_staker_operator_payout - floor(total_staker_operator_payout * COALESCE(oas.split, 1000) / 10000.0)
```

Comment:
Do we know what type the output of `COALESCE(oas.split, 1000) / 10000.0` is? We want the `total_staker_operator_payout * split` to be a `numeric * numeric` and not a `numeric * float`

Reply:
I mean the existing `0.10` is a float / double. So I've stuck with that. I believe the floor does round it down to a whole number.

Reply:
floor(total_staker_operator_payout * 0.10)` is `numeric * numeric` since Postgres treats constants as numerics: 
 
> A numeric constant that contains neither a decimal point nor an exponent is initially presumed to be type integer if its value fits in type integer (32 bits); otherwise it is presumed to be type bigint if its value fits in type bigint (64 bits); otherwise it is taken to be type numeric. Constants that contain decimal points and/or exponents are always initially presumed to be type numeric.

I just want to be really sure that the division is a numeric. `numeric/numeric` should be numeric? W

Reply:
Done in [f5ce3ef](https://github.com/Layr-Labs/sidecar/pull/117/commits/f5ce3ef0a51a7b482d2b9a0c74e377ed7c21f06b)